pafy/channel.py

```diff
@@ -0,0 +1,232 @@
+
+            for pl in playlistList['items']:
+                try:
+                    thumbnail = pl['snippet']['thumbnails']['standard']['url']
+                except KeyError:
+                    thumbnail = None
+                pl_data = dict(
+                    id=pl['id'],
+                    title=pl['snippet']['title'],
+                    author=pl['snippet']['channelTitle'],
+                    description=pl['snippet']['description'],
+                    thumbnail=thumbnail,
+                    len=pl['contentDetails']['itemCount']
+                )
+
+                pl_obj = Playlist.from_dict(pl_data, self._basic, self._gdata,
+                                            self._size, self._callback)
+                playlists.append(pl_obj)
+                if self._callback:
+                    self._callback("Added playlist: %s" % pl_data['title'])
+
+            if not playlistList.get('nextPageToken'):
+                break
+            query['pageToken'] = playlistList['nextPageToken']
+
+        self._playlists = playlists
+        return self._playlists
+
+    @property
+    def subscriptions(self):
+        if self._subscriptions is not None:
+            return self._subscriptions
+
+        subscriptions = []
+        query = {'part': 'snippet',
+                 'maxResults': 50,
+                 'channelId': self.channel_id}
+
+        while True:
+            subs_data = call_gdata('subscriptions', query)
+            sub_ids = []
+
+            for sub in subs_data['items']:
+                sub_ids.append(sub['snippet']['resourceId']['channelId'])
+
+            query2 = {'part': 'snippet, contentDetails, statistics',
+                      'id': ','.join(sub_ids),
+                      'maxResults': 50}
+
+            data = call_gdata('channels', query2)
+
+            for ch in data['items']:
+                channel_data = dict(
+                    id=ch['id'],
+                    title=ch['snippet']['title'],
+                    description=ch['snippet']['description'],
+                    logo=ch['snippet']['thumbnails']['default']['url'],
+                    subscriberCount=ch['statistics']['subscriberCount'],
+                    uploads=ch['contentDetails']['relatedPlaylists']['uploads']
+                )
+                sub_obj = Channel.from_dict(channel_data, self._basic,
+                                            self._gdata, self._size,
+                                            self._callback)
+                subscriptions.append(sub_obj)
+
+            if not subs_data.get('nextPageToken'):
+                break
+            query['pageToken'] = subs_data['nextPageToken']
+
+        self._subscriptions = subscriptions
+        return self._subscriptions
+
+    def __repr__(self):
+        if not self._have_basic:
+            self._fetch_basic()
+        keys = "Type Title Description SubscriberCount"
+        keys = keys.split(" ")
+        info = {"Type": "Channel",
```

Comment:
Are you using a separate `keys` variable just to preserve the order (rather than just iterating over `.items()`)?

It would probably be cleaner to just replace info with an array of tuples and remove `key`.